lib/api/objectPut.js

```diff
@@ -91,77 +90,68 @@ function _storeIt(bucket, objectKey, objMD, authInfo, canonicalID, cipherBundle,
-        });
-        return dataStore(objectKeyContext, cipherBundle, request, size,
-            streamingV4Params, backendInfo, log,
-            (err, dataGetInfo, calculatedHash) => {
-                if (err) {
-                    log.trace('error from data', {
-                        error: err,
-                        method: 'dataStore',
-                    });
-                    return callback(err);
-                }
-                // So that data retrieval information for MPU's and
-                // regular puts are stored in the same data structure,
-                // place the retrieval info here into a single element array
-                const dataGetInfoArr = [{
-                    key: dataGetInfo.key,
-                    size,
-                    start: 0,
-                    dataStoreName: dataGetInfo.dataStoreName,
-                }];
-                if (cipherBundle) {
-                    dataGetInfoArr[0].cryptoScheme =
-                        cipherBundle.cryptoScheme;
-                    dataGetInfoArr[0].cipheredDataKey =
-                        cipherBundle.cipheredDataKey;
-                }
-                metadataStoreParams.contentMD5 = calculatedHash;
-                return _storeInMDandDeleteData(
-                    bucketName, dataGetInfoArr, cipherBundle,
-                    metadataStoreParams, dataToDelete,
-                    logger.newRequestLoggerFromSerializedUids(
-                        log.getSerializedUids()), (err, contentMD5) => {
-                            if (err) {
-                                return callback(err);
-                            }
-                            pushMetric('putObject', log, {
-                                authInfo,
-                                bucket: bucketName,
-                                newByteLength: size,
-                                oldByteLength: prevContentLen,
-                            });
-                            return callback(null, contentMD5);
-                        });
-            });
-    }
-    log.trace('content-length is 0 so only storing metadata', {
-        method: 'services.metadataValidateAuthorization',
-    });
-    metadataStoreParams.contentMD5 = constants.emptyFileMd5;
-    const dataGetInfo = null;
-    return _storeInMDandDeleteData(bucketName, dataGetInfo, cipherBundle,
-        metadataStoreParams, dataToDelete,
-            logger.newRequestLoggerFromSerializedUids(log
-            .getSerializedUids()), (err, contentMD5) => {
-                if (err) {
-                    return callback(err);
-                }
-                pushMetric('putObject', log, {
-                    authInfo,
-                    bucket: bucketName,
-                    newByteLength: size,
-                    oldByteLength: prevContentLen,
+    // null - new object
+    // 0 or > 0 - existing object with content-length 0 or greater than 0
```

Comment:
they were left by accident in the refactor, they are supposed to explain the pushmetrics oldByteLength. I moved them to the right place